scripts/build/createTag.sh

```diff
@@ -0,0 +1,239 @@
+  echo
+  echo "Modes:"
+  echo "* 'files', only modify files (but don't commit changes)"
+  echo "* 'commit', modify files and commit changes (but don't tag)"
+  echo "* 'tag', modify files, commit and tag (but don't push)"
+  echo "* 'push', modify files, commit, tag and push to origin"
+  echo
+  exit $1
+}
+
+function checkValidInteger()
+{
+  if [ "$1" == "" ]; then
+    echo $0: at least one of the three tag tokens is empty
+    exit 1
+  fi
+
+  if ! [[ $1 =~ ^[0-9]+$ ]]; then
+    echo $0: tag token '"'${$1}'"' is not a number
+    exit 1
+  fi
+}
+
+#
+# Modify rpm/SPECS/contextBroker.spec only when step to a non-dev release
+# 
+# FIXME: unify this code with the same one in resease.sh into a commmon lib
+#
+function flushCNRToSpec()
+{
+  SPEC_FILE=$PROJECT_DIR/rpm/SPECS/contextBroker.spec
+
+  #
+  # Edit rpm/SPECS/contextBroker.spec, adding the new changes from CHANGELOG_FILE
+  #
+  # 1. Find the line in rpm/SPECS/contextBroker.spec, where to add the content of CHANGELOG_FILE plus the info-line for the changes.
+  #    o LINES:       number of lines before the insertion
+  # 2. Get the total number of lines in rpm/SPECS/contextBroker.spec
+  # 3. Get the number of lines in rpm/SPECS/contextBroker.spec after the insertion
+  #    o LAST_LINES:  number of lines after the insertion
+  # 4. To a temporal file, add the four 'chunks':
+  #    1. LINES
+  #    2. the info-line for the changes
+  #    3. the content of CHANGELOG_FILE
+  #    4. LAST_LINES
+  # 5. Replace using the temporal file
+
+  #
+  # 1. Find the line in rpm/SPECS/contextBroker.spec, where to add the content of CHANGELOG_FILE
+  #    The for is because these is more than one oceuurence of '%changelog'. We are only
+  #    interested in the last one.
+  #
+  for line in $(grep -n '%changelog' $SPEC_FILE | awk -F: '{ print $1 }')
+  do
+    LINE=$line
+  done
+
+  #
+  # 2. Get the total number of lines in rpm/SPECS/contextBroker.spec
+  #
+  LINES=$(wc -l $SPEC_FILE | awk '{ print $1 }')
+
+  #
+  # 3. Get the number of lines in rpm/SPECS/contextBroker.spec after the insertion
+  #
+  typeset -i LAST_LINES
+  LAST_LINES=$LINES-$LINE
+
+  #
+  # 4. To a temporal file, add the four 'chunks'
+  #
+  head -$LINE $SPEC_FILE        >  /tmp/contextBroker.spec
+
+  echo -n '* '                  >> /tmp/contextBroker.spec
+  echo $dateLine                >> /tmp/contextBroker.spec
+
+  cat $changelog                                  >> /tmp/contextBroker.spec
+  echo                                            >> /tmp/contextBroker.spec
+
+  tail -$LAST_LINES $SPEC_FILE  >> /tmp/contextBroker.spec
+    
+  # 5. Replace using the temporal file
+  mv /tmp/contextBroker.spec $SPEC_FILE
+
+  rm $changelog; touch $changelog
+}
+
+
+
+# ------------------------------------------------------------------------------
+#
+# Argument parsing
+#
+files=off
+commit=off
+tag=off
+push=off
+
+changelog=CHANGES_NEXT_RELEASE
+
+mode=""
+
+while [ "$#" != 0 ]
```

Comment:
Fixed in 0cfbfe7.

At the end, I changed to use short options (`-m`, `-u`, `-c`) as it seems that the alternative for long options (`getopt`, without `s`) is not portable.